### PR TITLE
update rivet, kakoune, add fastnlo, sherpa, mcgrid

### PR DESCRIPTION
**Pull request by veprbl**
###### Things done

- [ ] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [x] macOS
   - [ ] Linux
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/applications/science/physics/sherpa/default.nix [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    url = "http://www.hepforge.org/archive/sherpa/SHERPA-MC-${version}.tar.gz";
+    sha256 = "13vkz6w8kqyv8sgy3mxnlps5ykml5rnlj50vjj0pp9rgbl5y8ali";
+  };
+
+  buildInputs = [ gfortran sqlite ];
+
+  enableParallelBuilding = true;
+
+  meta = {
+    description = "Simulation of High-Energy Reactions of PArticles in lepton-lepton, lepton-photon, photon-photon, lepton-hadron and hadron-hadron collisions";
```
Comment (grahamc) [inline pkgs/applications/science/physics/sherpa/default.nix]: > PArticles

**Review by grahamc — CHANGES_REQUESTED** [review]
Just a little typo I think, but looks great. I love seeing these HEP tools in Nixpkgs.

File: pkgs/applications/science/physics/sherpa/default.nix (hunk 1 above)
Comment (veprbl) [inline pkgs/applications/science/physics/sherpa/default.nix]: S,H,E,R,PA :)

Comment (grahamc) [inline pkgs/applications/science/physics/sherpa/default.nix]: ... D,E,R,PA. Thank you :)

**Review by grahamc — APPROVED** [review]

**Pull request merged**